base/util.jl

```diff
@@ -68,13 +68,12 @@ gc_bytes() = ccall(:jl_gc_total_bytes, Int64, ())
 const _mem_units = ["byte", "KiB", "MiB", "GiB", "TiB", "PiB"]
 const _cnt_units = ["", " k", " M", " G", " T", " P"]
 function prettyprint_getunits(value, numunits, factor)
-    if value == 0 || value == 1
-        return (value, 1)
+    unit = 1
+    while value >= factor && unit < numunits
```

Comment:
This should be made type stable, e.g. by putting `value /= one(factor)` before the loop.

Reply:
When pretty-printing, it would be nice to omit the `.0` for exact numbers of bytes.  It seems weird to write `"998.0 bytes"`, for example.

Of course, that could be done in the printing routine rather than here, but I'm not sure type-stability is a big issue here.

We could always do it at the end by changing the return value to `return value == floor(value) ? Int(value) : value, unit`, which would also print e.g. 1024 bytes as "1 KiB" rather than "1.0 KiB".